client/src/index.ts

```diff
@@ -470,3 +479,30 @@ export class XataError extends Error {
 }
 
 export type Links = Record<string, Array<string[]>>;
+
+const isBranchStrategyBuilder = (strategy: BranchStrategy): strategy is BranchStrategyBuilder => {
+  return typeof strategy === 'function';
+};
+
+const isBranchStrategy = (strategy: BranchStrategy): strategy is BranchStrategy => {
+  return isBranchStrategyBuilder(strategy) || typeof strategy === 'string';
+};
+
+const getBranchFromStrategy = async (param: BranchStrategy | BranchStrategy[]): Promise<string | undefined | null> => {
+  const strategies = Array.isArray(param) ? param : [param];
+
+  const getBranch = async (strategy: BranchStrategy) => {
+    if (isBranchStrategyBuilder(strategy)) {
+      return await strategy();
+    } else if (isBranchStrategy(strategy)) {
+      return strategy;
+    } else {
+      throw new Error(`Invalid branch strategy: ${strategy}`);
+    }
+  };
+
+  const strategy = strategies.find((option) => getBranch(option) !== undefined);
+  if (strategy) return await getBranch(strategy);
+
+  return undefined;
+};
```

Comment:
Not 100% about code style preferences (type guards, typeof, helper methods and project structure, etc...), feel free to make comments about anything that should be different.